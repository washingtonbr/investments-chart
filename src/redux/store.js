import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import {persistStore, persistReducer} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import rootReducer from './reducers';

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const persistConfig = {
  key: 'root',
  storage: storage,
};
const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  {},
  composeEnhancer(applyMiddleware(thunk)),
);

export const persistor = persistStore(store);
