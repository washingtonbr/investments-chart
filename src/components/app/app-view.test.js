
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux'
import { mockReduxState } from '../../redux/store.mock';
import AppView from './app-view';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const reduxStore = mockStore(mockReduxState);

describe('AppView', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={reduxStore}>
          <MemoryRouter>
            <AppView />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
