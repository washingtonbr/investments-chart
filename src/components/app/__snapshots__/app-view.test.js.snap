// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView renders correctly 1`] = `
<div
  className="sc-AxjAm fYZJlX"
>
  <div
    className="sc-AxgMl jCMjGN"
  >
    <div
      className="sc-AxheI dUbsWX"
    >
      <div
        className="sc-AxirZ ejVXsm"
      >
        <select
          className="sc-AxiKw esaJYC"
          onChange={[Function]}
        />
      </div>
    </div>
    <div
      className="sc-AxhUy blYzjI"
    >
      Você está vendo o período 
      <strong>
        Desde o início
      </strong>
      .
    </div>
    Não há dados para o período selecionado.
  </div>
</div>
`;

exports[`Chart renders correctly 1`] = `
<div
  className="sc-AxjAm fYZJlX"
>
  <div
    className="sc-AxgMl jCMjGN"
  >
    <div
      className="sc-AxheI dUbsWX"
    >
      <div
        className="sc-AxirZ ejVXsm"
      >
        <select
          className="sc-AxiKw esaJYC"
          onChange={[Function]}
        />
      </div>
    </div>
    <div
      className="sc-AxhUy blYzjI"
    >
      Você está vendo o período 
      <strong>
        Desde o início
      </strong>
      .
    </div>
    Não há dados para o período selecionado.
  </div>
</div>
`;
